app/controllers/pieces_controller.rb

```diff
@@ -1,6 +1,9 @@
 class PiecesController < ApplicationController
+  before_action :authenticate_user!, only: [:update, :edit, :destroy]
+
   def new
     @piece = Piece.new
+    signed_in_flash
```

Comment:
This should be a controller callback (i.e. `before_action :signed_in_flash, only: :new)`